Source/DafnyDriver/DafnyDriver.cs

```diff
@@ -543,10 +566,17 @@ static string WriteDafnyProgramToFiles(string dafnyProgramName, string targetPro
           otherFiles.Add(wr.Filename, sw.ToString());
         }
       }
+      string baseName = Path.GetFileNameWithoutExtension(dafnyProgramName);
       string callToMain = null;
       if (hasMain) {
         using (var wr = new TargetWriter(0)) {
+          if (DafnyOptions.O.CompileTarget is DafnyOptions.CompilationTarget.Java) {
+            wr.Write("public class {0} {{\n\t", baseName);
```

Comment:
Do a WriteLine instead of using `\n`, because depending on the system, lines are ended with `\n`, `\r`, or `\r\n`.

Also, use spaces instead of tabs.